apyros/metaopen.py

```diff
@@ -0,0 +1,17 @@
+"""
+   only wrapper for ZIP files
+"""
+
+import os
+from zipfile import ZipFile
+
+
+def metaopen(filename, mode='r'):
+    """Wrapper for opening files optionally from ZIP"""
+    assert mode in ['r', 'rb'], mode  # just to know what modes we use
+    if '.zip' in filename:
+        return ZipFile.open(ZipFile(os.path.dirname(filename)), os.path.basename(filename))
```

Comment:
Sometimes I think you have a special gift for writing a code that works but one wonders a long time why... I know that in python you can call methods on objects with explicit self by calling it through the class, but... Nobody does that in regular code. I think the obvious way to call a method is this:

```python
return ZipFile(os.path.dirname(filename)).open(os.path.basename(filename))
```

There is nothing wrong with your code, it works correctly. It is just pretty non-obvious (I had to run it through the debugger just to find out what it does).